### PR TITLE
Issue 2359 get section by lang

### PR DESCRIPTION
**Pull request by SumDonkuS**
Adding optional `lang` argument for `get_section` global function.  Language code will be added to the filename if it is not the default language.  Related issue [here](https://github.com/getzola/zola/issues/2359).

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?

### CODE REVIEW

**Comment by Keats** [comment]
Thinking about it a bit more, we don't have it for `get_page` either... We should probably add it to both to be consistent and share the code when possible

File: components/templates/src/global_fns/content.rs [hunk 1]
```diff
@@ -119,19 +140,39 @@ impl TeraFn for GetSection {
...
-        let full_path = self.base_path.join(&path);
-        let library = self.library.read().unwrap();
-
-        match library.sections.get(&full_path) {
-            Some(s) => {
-                if metadata_only {
-                    Ok(to_value(s.serialize_basic(&library)).unwrap())
-                } else {
-                    Ok(to_value(s.serialize(&library)).unwrap())
+        let lang =
```
Comment (Keats) [inline components/templates/src/global_fns/content.rs:196]: We could also pass the list of available languages defined in config.toml and error if not in the list

Comment (SumDonkuS) [reply components/templates/src/global_fns/content.rs:196]: Thank you.  Added check for configured languages.  Let me know what you think.

**Comment by SumDonkuS** [comment]
Good call on adding it to `get_page` as well.  I have that added the code and tried to generalize as much code across to be used by both as possible.  Please let me know what you think.

File: components/templates/src/global_fns/content.rs [hunk 2]
```diff
@@ -71,14 +72,60 @@ impl TeraFn for GetTaxonomyUrl {
...
+                Err(format!("Error adding language code to {}", path).into())
+            } else {
+                Ok(Cow::Owned(format!("{}.{}{}", prefix.unwrap(), lang, suffix.unwrap())))
+            }
+        }
+        None => Ok(Cow::Owned(format!("{}.{}", path, lang))),
+    }
+}
+
+fn calculate_path<'a>(
```
Comment (Keats) [inline components/templates/src/global_fns/content.rs]: i'm not sure calculate_path is the right name. `get_path_with_lang`? Not sure

File: docs/content/documentation/templates/overview.md [hunk 3]
```diff
@@ -154,6 +160,12 @@ If you only need the metadata of the section, you can pass `metadata_only=true`
 {% set section = get_section(path="blog/_index.md", metadata_only=true) %}
 ```
 
+If selecting a specific language for the section, you can pass `lang` with the language code to the function:
```
Comment (Keats) [inline docs/content/documentation/templates/overview.md:169]: I would mention for pages and here that it's equivalent to `{% set section = get_section(path="blog/_index.fr.md") %}`

**Comment by SumDonkuS** [comment]
Great.  Changed the function name to be more specific, like you suggested.  Also, added to the docs equivalencies based on if the language passed is the default language or not.

**Review by Keats — APPROVED** [review]

**Comment by Keats** [comment]
Thanks!

**Pull request merged**